wkcuber/api/layer.py

```diff
@@ -560,43 +568,61 @@ def _setup_mag(self, mag: Union[str, Mag]) -> None:
         # This method is used to initialize the mag when opening the Dataset. This does not create e.g. the wk_header.
 
         # normalize the name of the mag
-        mag = Mag(mag).to_layer_name()
+        mag = Mag(mag)
+        mag_name = mag.to_layer_name()
 
         self._assert_mag_does_not_exist_yet(mag)
 
         try:
             with wkw.Dataset.open(
-                str(_find_mag_path_on_disk(self.dataset.path, self.name, mag))
+                str(_find_mag_path_on_disk(self.dataset.path, self.name, mag_name))
             ) as wkw_dataset:
                 wk_header = wkw_dataset.header
 
             self._mags[mag] = MagView(
-                self, mag, wk_header.block_len, wk_header.file_len, wk_header.block_type
+                self,
+                mag_name,
+                wk_header.block_len,
+                wk_header.file_len,
+                wk_header.block_type,
             )
 
             self.dataset.properties._add_mag(
-                self.name, mag, cube_length=wk_header.block_len * wk_header.file_len
+                self.name,
+                mag_name,
+                cube_length=wk_header.block_len * wk_header.file_len,
             )
         except wkw.WKWException:
             logging.error(
-                f"Failed to setup magnification {str(mag)}, which is specified in the datasource-properties.json"
+                f"Failed to setup magnification {mag_name}, which is specified in the datasource-properties.json"
             )
 
     def _initialize_mag_from_other_mag(
         self, new_mag_name: Union[str, Mag], other_mag: MagView, compress: bool
     ) -> MagView:
-        block_type = (
-            wkw.Header.BLOCK_TYPE_LZ4HC if compress else wkw.Header.BLOCK_TYPE_RAW
-        )
         return self.add_mag(
             new_mag_name,
             block_len=other_mag.header.block_len,
             file_len=other_mag.header.file_len,
-            block_type=block_type,
+            compress=compress,
         )
 
 
-class LayerTypes:
+class SegmentationLayer(Layer):
+    @property
+    def largest_segment_id(self) -> int:
+        layer = self.dataset.properties.data_layers[self.name]
+        assert isinstance(layer, SegmentationLayerProperties)
+        return layer.largest_segment_id
+
+    @largest_segment_id.setter
+    def largest_segment_id(self, largest_segment_id: int) -> None:
+        layer = self.dataset.properties._data_layers[self.name]
+        assert isinstance(layer, SegmentationLayerProperties)
+        layer._largest_segment_id = largest_segment_id
```

Comment:
I'd rename `layer` to `layer_properties`, and we should re-export the datasource-properties after setting largest_segment_id: 

```suggestion
    @property
    def largest_segment_id(self) -> int:
        layer_properties = self.dataset.properties.data_layers[self.name]
        assert isinstance(layer_properties, SegmentationLayerProperties)
        return layer_properties.largest_segment_id

    @largest_segment_id.setter
    def largest_segment_id(self, largest_segment_id: int) -> None:
        layer_properties = self.dataset.properties._data_layers[self.name]
        assert isinstance(layer_properties, SegmentationLayerProperties)
        layer_properties._largest_segment_id = largest_segment_id
        self.dataset.properties._export_as_json()
```